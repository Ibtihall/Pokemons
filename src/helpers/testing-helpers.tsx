import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { render, RenderOptions } from '@testing-library/react';
import store from '../store';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { renderHook } from '@testing-library/react-hooks';

export const Wrapper: React.FC = (props:any): JSX.Element => {
  return (
    <Provider store={store}>
      {//@ts-ignore
      <Router>
        {props.children} 
      </Router>}
    </Provider>
  );
};


export const renderWithIntl = (
  ui: ReactElement,
  renderOptions?: RenderOptions,
) => {
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export const renderHookWithIntl = <P, R>(hook: (props: P) => R) => {
  return renderHook(hook, { wrapper: Wrapper });
};
