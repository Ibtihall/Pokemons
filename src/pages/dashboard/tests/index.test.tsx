import { cleanup } from '@testing-library/react';
import { renderWithIntl } from '../../../helpers/testing-helpers';
import DashboardPage, { DashboardPageProps } from '..';
afterEach(cleanup);

describe('Pokemon DashboardPage', () => {
  let props: DashboardPageProps;
  beforeEach(() => {
    props = {
    };
 
  });
  describe('render', () => {
    it('should render correctly', () => {
      const { asFragment } = renderWithIntl(<DashboardPage {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
