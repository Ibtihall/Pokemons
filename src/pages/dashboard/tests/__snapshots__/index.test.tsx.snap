// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon DashboardPage render should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-jqUVSM dvldzU"
  >
    <div
      class="sc-kDDrLX brhNHo"
    >
      Who are you looking for?
    </div>
    <div>
      <input
        class="sc-iqcoie bHAfKa"
        placeholder="E.g Pikachu"
        type="search"
      />
      <button
        class="MuiButtonBase-root MuiIconButton-root sc-fLlhyt hbrTCC"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component "
      style="height: auto; overflow: auto;"
    >
      <div
        class="sc-fEOsli hXQQzM"
      />
      <div
        class="sc-fEOsli hXQQzM"
      >
        <div
          class="sc-bjUoiL xAOhY"
        >
          <h4>
            Loading...
          </h4>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
